src/integrations/authentication/implementations/openid/OpenID.ts

```diff
@@ -1,21 +1,36 @@
 
-import { AuthorizationParameters, BaseClient, ClientMetadata, Issuer } from 'openid-client';
-
-import { IdentityProvider } from '../../definitions/interfaces';
-import { Identity, Session } from '../../definitions/types';
-import NotConnected from '../../errors/NotConnected';
+import
+{
+    Configuration, DiscoveryRequestOptions,
+    allowInsecureRequests,
+    authorizationCodeGrant,
+    buildAuthorizationUrl,
+    calculatePKCECodeChallenge,
+    discovery,
+    fetchUserInfo,
+    randomPKCECodeVerifier,
+    refreshTokenGrant,
+    tokenRevocation
+} from 'openid-client';
+
+import { IdentityProvider } from '../../definitions/interfaces.js';
+import { Identity, Session } from '../../definitions/types.js';
+import NotConnected from '../../errors/NotConnected.js';
 
 type OpenIDConfiguration = {
     issuer: string;
     clientId: string;
     clientSecret: string;
     redirectUri: string;
+    allowInsecureRequests: boolean;
 };
 
 export default class OpenID implements IdentityProvider
 {
-    #configuration: OpenIDConfiguration;
-    #client?: BaseClient;
+    readonly #configuration: OpenIDConfiguration;
+    #config?: Configuration;
```

Comment:
Both names might be clear on their own, but combined their quite confusing...

```suggestion
    readonly #providerConfiguration: OpenIDConfiguration;
    #clientConfiguration?: Configuration;
```